lib/pharos/addon.rb

```diff
@@ -26,56 +26,70 @@ def self.messages
       end
     end
 
-    def self.name(name = nil)
-      if name
-        @name = name
-      else
-        @name
+    class << self
+      attr_writer :addon_location
+
+      # @param subclass [Class]
+      def inherited(subclass)
+        subclass.addon_location = File.dirname(caller(1..1).first[/(.+\.rb):\d+:in/, 1])
```

Comment:
This is a pretty awful hack, but I suppose it's well-enough contained that it's okay so long as it works for now.

Reply:
Idk if there is any other way to achieve this.

Reply:
If the addons were files using some DSL rather than using the standard `require`, then it would be part of the `AddonManager.load_addons(...)`... but that's a bit further out.

Reply:
DSL is now there.

Reply:
#361 isn't exactly what I meant by a DSL, it's still using `require`... but the `block.source_location.first` is slightly neater than the `caller(1..1).first`.